src/mach/pack/alpha/jib.clj

```diff
@@ -0,0 +1,103 @@
+(ns mach.pack.alpha.jib
+  (:require [mach.pack.alpha.impl.tools-deps :as tools-deps]
+            [clojure.string :as str]
+            [clojure.tools.cli :as cli])
+  (:import (com.google.cloud.tools.jib.api Jib AbsoluteUnixPath)
+           (java.nio.file Paths Files LinkOption)
+           (com.google.cloud.tools.jib.api LayerConfiguration
+                                           Containerizer
+                                           DockerDaemonImage
+                                           TarImage
+                                           RegistryImage
+                                           ImageReference)
+           (com.google.cloud.tools.jib.frontend CredentialRetrieverFactory)))
+
+(def string-array (into-array String []))
+(def target-dir "/home/app")
+
+(defn jib [{::tools-deps/keys [paths lib-map]} {:keys [image-name image-type tar-file base-image target-dir main]}]
+  (println "Building" image-name)
+  (-> (Jib/from base-image)
+      (.addLayer (into []
+                       (map #(Paths/get % string-array))
+                       (mapcat :paths (vals lib-map)))
+                 (AbsoluteUnixPath/get target-dir))
+      (.addLayer (-> (reduce (fn [acc project-path]
+                               (let [path (Paths/get project-path string-array)]
+                                 (if (Files/exists path (into-array LinkOption []))
+                                   (.addEntryRecursive acc
+                                                       path
+                                                       (AbsoluteUnixPath/get (str target-dir "/" project-path)))
+                                   acc)))
+                             (LayerConfiguration/builder)
+                             paths)
+                     (.build)))
+      (.setEntrypoint (into-array String ["java"
+                                          "-cp" (str/join ":" (cons (str target-dir "/*")
```

Comment:
Unfortunately, using "*" only utilizes .jar files, and ignores directories (which would be brought in by git dependencies).

This is a frustrating limitation I've run into before.

Reply:
Nice, thanks for the tip! :)